tony-core/src/main/java/com/linkedin/tony/LocalizableResource.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright 2019 LinkedIn Corporation. All rights reserved. Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.tony;
+
+import org.apache.commons.cli.ParseException;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.yarn.api.records.LocalResource;
+import org.apache.hadoop.yarn.api.records.LocalResourceType;
+import org.apache.hadoop.yarn.api.records.LocalResourceVisibility;
+import org.apache.hadoop.yarn.util.ConverterUtils;
+
+import java.io.IOException;
+import java.net.URI;
+
+/**
+ * Parse container resources of format:
+ * SOURCE_FILE_PATH::PATH_IN_CONTAINER#archive
+ * only SOURCE_FILE_PATH is required.
+ *
+ * SOURCE_FILE_PATH: location of the file to be localized to containers. This could be either local resources or remote
+ *                   resources.
+ * FILE_IN_CONTAINER: optional, default to source file name. If specified, source_file_path will be localized as name
+ *                   file_in_container in container.
+ * ARCHIVE: if #archive is put at the end, the file will be uploaded as ARCHIVE type and unzipped upon localization.
+ */
+public class LocalizableResource {
+
+    private String path;
+
+    private boolean isDirectory;
+    private LocalResourceType resourceType;
+    private FileStatus sourceFileStatus;
+    private Path sourceFilePath;
+    private String localFileName;
+
+    public boolean isDirectory() {
+        return isDirectory;
+    }
+
+    public Path getSourceFilePath() {
+        return sourceFilePath;
+    }
+
+    public String getLocalFileName() {
+        return localFileName;
+    }
+
+    public LocalizableResource(String path) {
+        this.path = path;
+    }
+
+    public void parse(FileSystem fs) throws ParseException, IOException {
+        String filePath = path;
+        resourceType = LocalResourceType.FILE;
+        if (path.contains(Constants.ARCHIVE_SUFFIX)) {
+            resourceType = LocalResourceType.ARCHIVE;
+            filePath = path.replace(Constants.ARCHIVE_SUFFIX, "");
+        }
+        String[] tuple = filePath.split(Constants.RESOURCE_DIVIDER);
+        if (tuple.length > 2) {
+            throw new ParseException("Failed to parse file: " + path);
+        }
+
+        sourceFilePath = new Path(tuple[0]);
+        sourceFileStatus = fs.getFileStatus(sourceFilePath);
+        localFileName = sourceFilePath.getName();
+        if (tuple.length == 2) {
+            localFileName = tuple[1];
+        }
+
+        if (sourceFileStatus.isDirectory()) {
+            isDirectory = true;
+        }
+    }
+
+    public LocalResource getLocalResource() {
```

Comment:
`toLocalResource()` sounds more accurate to me (you're converting from one representation to another).